packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -737,9 +737,16 @@ public void setEditable(ReactEditText view, boolean editable) {
 
   @ReactProp(name = ViewProps.NUMBER_OF_LINES, defaultInt = 1)
   public void setNumLines(ReactEditText view, int numLines) {
+    view.setInputType(view.getInputType() | InputType.TYPE_TEXT_FLAG_MULTI_LINE);
     view.setLines(numLines);
   }
 
+  @ReactProp(name = ViewProps.MAXIMUM_NUMBER_OF_LINES, defaultInt = 1)
+  public void setMaxNumLines(ReactEditText view, int numLines) {
+    view.setInputType(view.getInputType() | InputType.TYPE_TEXT_FLAG_MULTI_LINE);
```

Comment:
```suggestion
    if ((view.getInputType() & InputType.TYPE_TEXT_FLAG_MULTI_LINE) == 0) {
      view.setInputType(view.getInputType() | InputType.TYPE_TEXT_FLAG_MULTI_LINE);
    }
```

We've seen some weird text input flickering on the Expensify app when building for the new arch when the flag is being set on every render.

Reply:
That makes a lot of sense. Thanks for info and quick fix :)